app/uk/gov/hmrc/exports/services/notifications/NotificationHelper.scala

```diff
@@ -0,0 +1,120 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.services.notifications
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+import scala.xml.XML
+
+import javax.inject.{Inject, Singleton}
+import play.api.Logging
+import uk.gov.hmrc.exports.connectors.{CustomsDataStoreConnector, EmailConnector}
+import uk.gov.hmrc.exports.models.declaration.notifications.{Notification, NotificationDetails}
+import uk.gov.hmrc.exports.models.declaration.submissions.{Submission, SubmissionStatus}
+import uk.gov.hmrc.exports.models.emails.SendEmailResult._
+import uk.gov.hmrc.exports.models.emails._
+import uk.gov.hmrc.exports.repositories.{NotificationRepository, SubmissionRepository}
+import uk.gov.hmrc.http.HeaderCarrier
+
+@Singleton
+class NotificationHelper @Inject()(
+  customsDataStoreConnector: CustomsDataStoreConnector,
+  emailConnector: EmailConnector,
+  notificationRepository: NotificationRepository,
+  submissionRepository: SubmissionRepository
+)(implicit executionContext: ExecutionContext)
+    extends Logging {
+
+  def parseAndSave(unparsed: Notification)(implicit hc: HeaderCarrier): Future[Unit] =
+    Future {
+      Try(XML.loadString(unparsed.payload)).map(NotificationParser.parse) match {
+        case Success(notificationDetails) if notificationDetails.nonEmpty =>
+          saveParsedNotifications(unparsed, notificationDetails)
+
+        case Success(_) => ()
+
+        case Failure(exc) =>
+          logger.warn(s"There was a problem while parsing notification with actionId=${unparsed.actionId}. Error was: ${exc.getMessage}")
+      }
+    }
+
+  private def sendEmailForDmsdocNotification(emailAddress: VerifiedEmailAddress, mrn: String, eori: String)(implicit hc: HeaderCarrier): Unit = {
+    val emailParameters = EmailParameters(Map(EmailParameter.MRN -> mrn))
+    val sendEmailRequest = SendEmailRequest(List(emailAddress.address), TemplateId.DMSDOC_NOTIFICATION, emailParameters)
+    emailConnector.sendEmail(sendEmailRequest).map {
+      case EmailAccepted                      =>
+      case BadEmailRequest(message)           => logSendEmailError("Bad request", message, eori, mrn)
+      case InternalEmailServiceError(message) => logSendEmailError("Email service error", message, eori, mrn)
+    }
+  }
+
+  private def sendEmailIfDmsdocNotification(details: NotificationDetails, submission: Submission)(implicit hc: HeaderCarrier): Boolean = {
```

Comment:
Does this need to return a Boolean if it will always be true? Probably better to return Unit